src/System.Private.CoreLib/shared/System/Number.BigInteger.cs

```diff
@@ -198,6 +406,217 @@ public static int Compare(ref BigInteger lhs, ref BigInteger rhs)
                 return 0;
             }
 
+            public static uint CountSignificantBits(uint value)
+            {
+                return (value != 0) ? (1 + LogBase2(value)) : 0;
+            }
+
+            public static uint CountSignificantBits(ulong value)
+            {
+                uint upper = (uint)(value >> 32);
+
+                if (upper != 0)
+                {
+                    return 32 + CountSignificantBits(upper);
+                }
+
+                return CountSignificantBits((uint)(value));
+            }
+
+            public static uint CountSignificantBits(ref BigInteger value)
+            {
+                if (value.IsZero())
+                {
+                    return 0;
+                }
+
+                // We don't track any unused blocks, so we only need to do a BSR on the
+                // last index and add that to the number of bits we skipped.
+
+                uint lastIndex = (uint)(value._length - 1);
+                return (lastIndex * BitsPerBlock) + CountSignificantBits(value._blocks[lastIndex]);
+            }
+
+            public static void DivRem(ref BigInteger lhs, ref BigInteger rhs, out BigInteger quo, out BigInteger rem)
+            {
+                // This is modified from the CoreFX BigIntegerCalculator.DivRem.cs implementation:
```

Comment:
Probably some room for improvements here. We only care about the lower 64-bits of `quo`, for example; and `rem` only matters if it is zero or non-zero.